example/lib/bloc/airship_bloc.dart

```diff
@@ -0,0 +1,154 @@
+import 'dart:async';
+import 'package:airship_example/bloc/bloc_base.dart';
+import 'package:airship/airship.dart';
+import 'package:flutter/material.dart';
+
+class AirshipBloc implements BlocBase {
+  final _channelStreamController = StreamController<String>();
+  final _channelUpdatedStreamController = StreamController<String>();
+
+  final _namedUserStreamController = StreamController<String>.broadcast();
+  final _namedUserSetStreamController = StreamController<String>();
+
+  final _notificationsEnabledStreamController = StreamController<bool>.broadcast();
+  final _notificationsEnabledSetStreamController = StreamController<bool>();
+
+  final _tagsStreamController = StreamController<List<String>>.broadcast();
+  final _tagsAddedStreamController = StreamController<List<String>>();
+  final _tagsRemovedStreamController = StreamController<List<String>>();
+
+  final _messagesStreamController = StreamController<List<InboxMessage>>.broadcast();
+  final _messageAddedStreamController = StreamController<InboxMessage>();
+  final _messageRemovedStreamController = StreamController<InboxMessage>();
+
+  Stream<String> get channelStream => _channelStreamController.stream;
+  StreamSink<String> get channelSink => _channelStreamController.sink;
+
+  Stream<String> get namedUserStream => _namedUserStreamController.stream;
+  StreamSink<String> get namedUserSink=> _namedUserStreamController.sink;
+  StreamSink<String> get namedUserSetSink => _namedUserSetStreamController.sink;
+
+  Stream<bool> get notificationsEnabledStream => _notificationsEnabledStreamController.stream;
+  StreamSink<bool> get notificationsEnabledSink => _notificationsEnabledStreamController.sink;
+  StreamSink<bool> get notificationsEnabledSetSink => _notificationsEnabledSetStreamController.sink;
+
+  Stream<List<String>> get tagsStream => _tagsStreamController.stream;
+  StreamSink<List<String>> get tagsSink => _tagsStreamController.sink;
+
+  Stream<List<InboxMessage>> get messagesStream => _messagesStreamController.stream;
+  StreamSink<List<InboxMessage>> get messagesSink => _messagesStreamController.sink;
+
+  StreamSink<List<String>> get tagsAddedSink => _tagsAddedStreamController.sink;
+  StreamSink<List<String>> get tagsRemovedSink => _tagsRemovedStreamController.sink;
+
+  StreamSink<InboxMessage> get messageRemovedSink => _messageRemovedStreamController.sink;
+
+  AirshipBloc() {
+    Airship.channelId.then((val) {
+      _channelStreamController.add(val);
+    });
+
+    Airship.tags.then((val) {
+      _tagsStreamController.add(val);
+    });
+
+    _tagsAddedStreamController.stream.listen((tags) {
```

Comment:
messagesSteam is public, tagsAddedSteam is not. Streams that are accessed outside of this class are given a blahStream property, otherwise it's fine to access the streamController directly to get at its stream property.

Reply:
OK